.github/workflows/tests.yml

```diff
@@ -6,17 +6,39 @@ on:
   push:
     branches:
       - main
-
+permissions:
```

Comment:
Yeah, I could not decide. 

I ended up with this flavor, because I need to [generate `dist/*`](https://github.com/aws-actions/aws-secretsmanager-get-secrets/blob/feature/integration-tests-in-pull-request/.github/workflows/tests.yml#L24) folder and not rely on author of PR providing it. When I split it into two separate workflows, I need to run [npm ci](https://github.com/aws-actions/aws-secretsmanager-get-secrets/blob/feature/integration-tests-in-pull-request/.github/workflows/tests.yml#L21) for each checkout separately. Its probably not a big deal, but I used it as reasoning for having integ tests with the rest of original workflow.

If you have some data based opinion, let me know and I'll be happy to split it.